proposals/NNNN-SerialExecutor-isIsolated.md

```diff
@@ -125,37 +128,15 @@ This proposal specifically adds the "if `isIsolatingCurrentContext` is available
 
-### Enabling `isIsolatingCurrentContext` checking mode
-
-Similar as complex equality in `SerialExecutors` this feature must be opted into by flagging it when the `UnownedSerialExecutor` value is returned from a serial executor's `asUnownedSerialExecutor()`.
-
-Previously this was done by signalling the feature in the `UnownedSerialExecutor` initializer like this:
-
-```swift
-// Existing API
-public func asUnownedSerialExecutor() -> UnownedSerialExecutor {
-  UnownedSerialExecutor(complexEquality: self)
-}
-```
-
-Which enables the following `isSameExclusiveExecutionContext` check, which can only be used when a "current" executor is present, and cannot be used when running code outside of a Swift concurrency task (!):
+### Detecting the `isIsolatingCurrentContext` checking mode
 
-```swift
-// Existing API
-public func isSameExclusiveExecutionContext(other: NaiveQueueExecutor) -> Bool {
-  other.secretIdentifier == self.secretIdentifier
-}
-```
+The `isIsolatingCurrentContext` method effectively replaces the `checkIsolated` method, because it can answer the same question _if it is implemented_.
 
-In order to enable the runtime to call into the `isIsolatingCurrentContext` the `UnownedSerialExecutor` **must** be constructed as follows:
+Some runtimes may not be able to implement a the returning `isIsolatingCurrentContext`, and they are not expected to implement the new protocol requirement.
 
-```swift
-public func asUnownedSerialExecutor() -> UnownedSerialExecutor {
-  UnownedSerialExecutor(hasIsIsolatingCurrentContext: self)
-}
-```
+The general guidance about which method to implement is to implement `isIsolatingCurrentContext` whenever possible. This method can be used by the Swift runtime in "warning mode". When running a check in this mode, the `checkIsolated` method cannot and will not be used because it would cause an unexpected crash.
 
-This sets a flag inside the internal executor reference which makes the swift runtime call into the new `isIsolatingCurrentContext` function, rather than the other versions of isolation checking. In many ways this API is the most general of them all, and generally preferable _if_ your executor is using some kind of mechanism to track the "current" context that the Swift concurrency runtime cannot know about, e.g. like thread local values inside threads managed by your executor.
+The presence of a non-default implementation of the `isIsolatingCurrentContext` protocol requirement. In other words, if there is an implementation of the requirement available _other than_ the default one provided in the concurrency library, the runtime will attempt to use this method _over_ the `checkIsolated` API. This allows for a smooth migration to the new API, and enables the use of this method in if the runtime would like issue a check that cannot cause a crash.
```

Comment:
The first sentence is missing... something

Reply:
Thank you, fixed!

> The presence of a non-default implementation of the `isIsolatingCurrentContext` protocol witness is detected by the compiler and the runtime can detect this information in order to determine if the new function should be used for these checks.